lib/indexeddb.js

```diff
@@ -0,0 +1,118 @@
+// Implemented by window & worker
+declare interface IDBEnvironment {
+  indexedDB: IDBFactory;
+}
+
+type IDBDirection = 'next' | 'nextunique' | 'prev' | 'prevunique';
+
+// Implemented by window.indexedDB & worker.indexedDB
+declare interface IDBFactory {
+  open(name: string, version?: number): IDBOpenDBRequest;
+  deleteDatabase(name: string): IDBOpenDBRequest;
+  cmp(a: any, b: any): -1|0|1;
+}
+
+declare interface IDBRequest extends EventTarget {
```

Comment:
It's too bad we don't support unions for interfaces. It would be cool to write

```JavaScript
declare interface IDBRequest extends EventTarget {
  readState: 'pending',
  error: Error,
  result: IDBObjectStore;
  ...
} | {
  readyState: 'done';
  error: Error;
  result: void;
} | {
  readyState: 'done',
  error: null,
  result: IDBObjectStore;
  ...
}

As written, this interface doesn't reflect the fact that `result` may be null and `error` may be undefined. Not a huge deal, but maybe worth representing in the interface.